Sources/PointFree/Auth.swift

```diff
@@ -128,6 +136,40 @@ extension EitherIO {
   }
 }
 
+private func fetchOrRegisterUser(env: GitHub.UserEnvelope) -> EitherIO<Prelude.Unit, Database.User> {
+
+  return AppEnvironment.current.database.fetchUserByGitHub(env.accessToken)
+    .flatMap { user in
+      EitherIO(run: IO { user.map(Either.right) ?? .left(unit) })
+        .catch(const(registerUser(env: env)))
```

Comment:
catching the error in here means we didnt find the user (as opposed to having a sql error)